cmake/EigenTesting.cmake

```diff
@@ -19,7 +19,9 @@ macro(ei_add_test_internal testname testname_with_suffix)
   endif()
 
   if(EIGEN_ADD_TEST_FILENAME_EXTENSION STREQUAL cu)
-    if(EIGEN_TEST_CUDA_CLANG)
+    if(EIGEN_TEST_HIP)
+      hip_add_executable(${targetname} ${filename} HIPCC_OPTIONS "-DEIGEN_USE_HIP")
+    elseif(EIGEN_TEST_CUDA_CLANG)
       set_source_files_properties(${filename} PROPERTIES LANGUAGE CXX)
```

Comment:
This question is due to my lack of HIP experience.  Do we not need to tell CMake that ${filename} is C++ as well?

Reply:
we do not seem to need to do that...don't know why.

@whchung who would be the right person to ask about `hip_add_executable` works (defined in `/opt/rocm/hip/cmake/FindHIP.cmake`. I had to collapase all subtests into one (i.e only use CALL_SUBTEST instead of CALL_SUBTEST_1, CALL_SUBTEST_2, ....) because passing in HIPCC_OPTIONS does not quite work as expected. This was resulting in strange behaviours when running the tests until I realized what was going on and decided to collapse them all into one.

Reply:
Talk to Maneesh, he's the owner of FindHIP.cmake